src/daemon/XorgDisplayServer.cpp

```diff
@@ -88,28 +89,55 @@ namespace SDDM {
     }
 
     void XorgDisplayServer::addCookie(const QString &file) {
-        // log message
+        Xauth auth = { 0 };
```

Comment:
This will only initialize the first member.

Reply:
It will value-initalize the remaining members. This is an [aggregate initializer](https://en.cppreference.com/w/cpp/language/aggregate_initialization): "If the number of initializer clauses is less than the number of members [...] or initializer list is completely empty, the remaining members [...] are initialized [...] by empty lists, in accordance with the usual list-initialization rules (which performs value-initialization for non-class types and non-aggregate classes with default constructors, and aggregate initialization for aggregates)." [Value initialization](https://en.cppreference.com/w/cpp/language/value_initialization) means "if T is an array type, each element of the array is value-initialized; otherwise, the object is zero-initialized."

The same effect could be achieved with `Xauth auth{};` which is not the same as `Xauth auth;` — the latter does [default initialization](https://en.cppreference.com/w/cpp/language/default_initialization): when no default constructor is available, "nothing is done: the objects with automatic storage duration (and their subobjects) are initialized to indeterminate values."

Reply:
C++ has too many initialization methods - the "unified initialization" just made everything more complicated...

I'd very much prefer `Xauth auth{};` here.